stratosphere/pm/source/impl/pm_process_manager.cpp

```diff
@@ -0,0 +1,680 @@
+            }
+        }
+
+        /* Process Tracking globals. */
+        HosThread g_process_track_thread;
+        SessionManagerBase *g_process_waitable_manager = nullptr;
+
+        /* Process lists. */
+        ProcessList g_process_list;
+        ProcessList g_dead_process_list;
+
+        /* Global events. */
+        IEvent *g_process_event = nullptr;
+        IEvent *g_hook_to_create_process_event = nullptr;
+        IEvent *g_hook_to_create_application_process_event = nullptr;
+        IEvent *g_boot_finished_event = nullptr;
+
+        /* Process Launch synchronization globals. */
+        IEvent *g_process_launch_start_event = nullptr;
+        HosSignal g_process_launch_finish_signal;
+        Result g_process_launch_result = ResultSuccess;
+        LaunchProcessArgs g_process_launch_args = {};
+
+        /* Hook globals. */
+        std::atomic<ncm::TitleId> g_title_id_hook;
+        std::atomic<bool> g_application_hook;
+
+        /* Helpers. */
+        void ProcessTrackingMain(void *arg) {
+            /* This is the main loop of the process tracking thread. */
+
+            /* Create waitable manager. */
+            static auto s_process_waiter = WaitableManager(1);
+            g_process_waitable_manager = &s_process_waiter;
+
+            /* Service processes. */
+            g_process_waitable_manager->AddWaitable(g_process_launch_start_event);
+            g_process_waitable_manager->Process();
+        }
+
+        inline u32 GetLoaderCreateProcessFlags(u32 launch_flags) {
+            u32 ldr_flags = 0;
+
+            if (ShouldSignalOnException(launch_flags) || (GetRuntimeFirmwareVersion() >= FirmwareVersion_200 && !ShouldStartSuspended(launch_flags))) {
+                ldr_flags |= ldr::CreateProcessFlag_EnableDebug;
+            }
+            if (ShouldDisableAslr(launch_flags)) {
+                ldr_flags |= ldr::CreateProcessFlag_DisableAslr;
+            }
+
+            return ldr_flags;
+        }
+
+        Result StartProcess(std::shared_ptr<ProcessInfo> process_info, const ldr::ProgramInfo *program_info) {
+            R_TRY(svcStartProcess(process_info->GetHandle(), program_info->main_thread_priority, program_info->default_cpu_id, program_info->main_thread_stack_size));
+            process_info->SetState(ProcessState_Running);
+            return ResultSuccess;
+        }
+
+        Result LaunchProcess(const LaunchProcessArgs *args) {
+            /* Get Program Info. */
+            ldr::ProgramInfo program_info;
+            R_TRY(ldr::pm::GetProgramInfo(&program_info, args->location));
+            const bool is_application = (program_info.flags & ldr::ProgramInfoFlag_ApplicationTypeMask) == ldr::ProgramInfoFlag_Application;
+            const bool allow_debug    = (program_info.flags & ldr::ProgramInfoFlag_AllowDebug) || GetRuntimeFirmwareVersion() < FirmwareVersion_200;
+
+            /* Ensure we only try to run one application. */
+            if (is_application) {
```

Comment:
Uh... is this correct? Shouldn't it be `if (is_application && /* already has application */) {` ?

Reply:
Haha, yeah, good catch.